sdk/resourcemanager/docs/RBAC.md

```diff
@@ -0,0 +1,117 @@
+
+Managed identity provides an identity for the Azure resource in Azure Active Directory, and uses it to obtain Azure AD token.
+RBAC enforces the role, scope, and access control of that managed identity. 
+
+They are useful in scenarios of enabling Azure virtual machine or web app to either manage Azure resource, or access data in Azure resource.
+
+The role of Owner is required for assigning role to managed identity. It can be configured from Azure Active Directory blade in Portal, or by Azure CLI.
+
+## Managing Azure resource
+
+Sample code to enable system-assigned managed identity to a virtual machine, and give it the role of contributor to a resource group.
+```java
+virtualMachine.update()
+    .withSystemAssignedManagedServiceIdentity()
+    .withSystemAssignedIdentityBasedAccessTo(resourceGroup.id(), BuiltInRole.CONTRIBUTOR)
+    .apply();
+```
+
+It allows code running in this particular virtual machine to manage other Azure resource on the target resource group.
+
+Managed identity can be enabled with similar code on Azure web app.
+
+SDK will support managed identity on Azure Kubernetes in the future.
+
+## Accessing data in Azure resource
+
+### Storage account
+
+Sample code to enable system-assigned managed identity to a web app, and give it the role of blob data contributor to a storage account.
+```java
+webApp.update()
+    .withSystemAssignedManagedServiceIdentity()
+    .withSystemAssignedIdentityBasedAccessTo(storageAccount.id, BuiltInRole.STORAGE_BLOB_DATA_CONTRIBUTOR)
+    .apply();
+```
+
+It allows code running in this particular web app to access blob data in the target storage account.
+
+### Key vault
+
+RBAC on key vault is in preview. SDK will support this feature in the future.
+
+For now, SDK supports access policies for key vault.
+
+Sample code to allow access of secret to user-assigned managed identity.
+```java
+vault.update()
+    .defineAccessPolicy()
+        .forObjectId(identity.principalId())
+        .allowSecretPermissions(SecretPermissions.GET)
+        .attach()
+    .apply();
+```
+
+For system-assigned managed identity, use e.g. `virtualMachine.systemAssignedManagedServiceIdentityPrincipalId()` in place of `identity.principalId()`.
+
+## Accessing Azure Active Directory
+
+SDK provides limited support for accessing Azure Active Directory, generally only for querying applications, users, groups, and service principals.
+
+This requires additional permission from Azure Active Directory, which can be configured from Azure Active Directory blade in Portal, or by Azure CLI.
+
+Since 2.2.0, SDK switched from [Azure Active Directory Grpah API][aad_graph] to [Microsoft Graph API][microsoft_graph].
+
+Permission required:
+- Access application and service principal: Microsoft Graph, Application permission, Application.Read.All
+- Access user: Microsoft Graph, Application permission, User.Read.All
+- Access group: Microsoft Graph, Application permission, Group.Read.All
```

Comment:
Could relate to more details like https://docs.microsoft.com/en-us/graph/api/user-list#permissions